test/route.go

```diff
@@ -63,24 +64,48 @@ func UpdateBlueGreenRoute(logger *logging.BaseLogger, clients *Clients, names, b
 	return clients.ServingClient.Routes.Patch(names.Route, types.JSONPatchType, patchBytes, "")
 }
 
+// Prober is the interface for a prober, which checks the result of the probes when stopped.
+type Prober interface {
+	// Stop terminates the prober, and checks for errors, surfacing anything
+	// error-worthy as test errors.
+	Stop(t *testing.T)
+}
+
+type prober struct {
+	logger *logging.BaseLogger
+	errCh  chan error
+}
+
+// prober implements Prober
+var _ Prober = (*prober)(nil)
+
+// Stop implements Prober
+func (p *prober) Stop(t *testing.T) {
+	defer close(p.errCh)
+	select {
+	case err := <-p.errCh:
+		t.Errorf("Prober encountered an error: %v", err)
+	default:
+		p.logger.Info("No error happens in the Route prober.")
+	}
+}
+
 // RunRouteProber creates and runs a prober as background goroutine to keep polling Route.
 // It stops when getting an error response from Route.
-func RunRouteProber(logger *logging.BaseLogger, clients *Clients, domain string) <-chan error {
+func RunRouteProber(logger *logging.BaseLogger, clients *Clients, domain string) Prober {
 	logger.Infof("Starting Route prober for route domain %s.", domain)
-	errorChan := make(chan error, 1)
+	p := &prober{logger: logger, errCh: make(chan error, 1)}
```

Comment:
probably would make a good debugging tool to do
`logger = logger.With()` the domain and then in Stop refer to the domain.